Steamfitter.Api/Services/TaskExecutionService.cs

```diff
@@ -406,12 +412,15 @@ private async STT.Task<Data.TaskStatus> ProcessTaskAsync(TaskEntity taskToExecut
             foreach (var resultEntity in resultEntityList)
             {
                 resultEntity.InputString = resultEntity.InputString.Replace("{moid}", resultEntity.VmId.ToString());
-                resultEntity.VmName = _stackStormService.GetVmName((Guid)resultEntity.VmId);
+                if (taskToExecute.ApiUrl != "http")
```

Comment:
Sorry if this is completely the wrong flow, but should this be checking for `requiresVm` instead of the ApiUrl?